cmd/snap-confine/mount-support.c

```diff
@@ -878,6 +903,47 @@ static bool __attribute__((used))
 	return false;
 }
 
+static struct sc_mount *sc_homedir_mounts(const struct sc_invocation *inv)
+{
+	if (inv->homedirs == NULL) {
+		return NULL;
+	}
+
+	int num_homedirs = 0;
+	for (char **path = inv->homedirs; *path != NULL; path++) {
+		num_homedirs++;
+	}
+
+	// We add one element for the end-of-array indicator.
+	struct sc_mount *mounts = calloc(num_homedirs + 1, sizeof(struct sc_mount));
+	if (mounts == NULL) {
+		die("cannot allocate mount data for homedirs");
+	}
+
+    // Copy inv->homedirs to the mount structures
+	for (int i = 0; i < num_homedirs; i++) {
+		debug("Adding homedir: %s", inv->homedirs[i]);
+		mounts[i].path = sc_strdup(inv->homedirs[i]);
```

Comment:
Should `is_bidirectional` also be set here? I assume changes should be propagated back to the real home.

Reply:
Really good point. Code updated.